components/org.wso2.carbon.identity.idle.account.identification/src/main/java/org/wso2/carbon/identity/idle/account/identification/services/impl/IdleAccountIdentificationServiceImpl.java

```diff
@@ -0,0 +1,174 @@
+            List<String> usernames = IdleAccountIdentificationDataHolder.getInstance().getIdentityDataStoreService()
+                    .getUserNamesBetweenProvidedClaimValues(IdleAccIdentificationConstants.LAST_LOGIN_TIME_CLAIM,
+                            excludeDateEpoch, inactiveDateEpoch, tenantId);
+            if (!usernames.isEmpty()) {
+                inactiveUsers = buildInactiveUsers(usernames);
+            }
+        } catch (IdentityException e) {
+            IdleAccIdentificationConstants.ErrorMessages errorEnum =
+                    IdleAccIdentificationConstants.ErrorMessages.ERROR_RETRIEVE_INACTIVE_USERS_FROM_DB;
+            throw new IdleAccountIdentificationServerException(errorEnum.getCode(), errorEnum.getMessage());
+        }
+        return inactiveUsers;
+    }
+
+    /**
+     * Build a list of inactive users.
+     *
+     * @param usernames        list of usernames.
+     * @return                 list of inactive user objects.
+     */
+    private List<InactiveUserModel> buildInactiveUsers(List<String> usernames)
+            throws IdleAccountIdentificationServerException {
+
+        List<InactiveUserModel> inactiveUsers = new ArrayList<>();
+        for (String username : usernames) {
+            String userId = fetchUserId(username);
+            if (StringUtils.isNotBlank(userId)) {
+                InactiveUserModel inactiveUser = new InactiveUserModel();
+                inactiveUser.setUsername(username);
+                inactiveUser.setUserId(userId);
+                inactiveUser.setUserStoreDomain(UserCoreUtil.extractDomainFromName(username));
+                inactiveUsers.add(inactiveUser);
+            }
+        }
+        return inactiveUsers;
+    }
+
+    /**
+     * Fetch UUID of the user.
+     *
+     * @param username          username of the user.
+     * @return                  UUID of the user
+     */
+    public String fetchUserId(String username) throws IdleAccountIdentificationServerException {
+
+        UserStoreManager userStoreManager = getUserStoreManager(UserCoreUtil.extractDomainFromName(username));
+        try {
+            if (userStoreManager instanceof UniqueIDJDBCUserStoreManager) {
+                return ((UniqueIDJDBCUserStoreManager) userStoreManager).getUserIDFromUserName(username);
+            }
+        } catch (UserStoreException e) {
+            IdleAccIdentificationConstants.ErrorMessages errorEnum =
+                    IdleAccIdentificationConstants.ErrorMessages.ERROR_RETRIEVE_USER_UUID;
+            throw new IdleAccountIdentificationServerException(errorEnum.getCode(), errorEnum.getMessage());
+        }
+        return null;
+    }
+
+    /**
+     * Get user store manager.
+     *
+     * @return UserStoreManager.
+     * @throws IdleAccountIdentificationServerException Exception when getting user store manager.
+     */
+    private UserStoreManager getUserStoreManager(String userStore) throws IdleAccountIdentificationServerException {
```

Comment:
Param name `userStore` starts to confuse when you continue to read the implementation of the method. Change it to reflect that this is the user store domain name.

Reply:
Addressed with commit c008575c8f41210bfec80da3de58500170ad968d